src/plugins/orangehrmRecruitmentPlugin/Api/CandidateAPI.php

```diff
@@ -0,0 +1,639 @@
+            $this->getFromDateFilterParam()
+        );
+        $candidateSearchFilterParamHolder->setToDate(
+            $this->getToDateFilterParam()
+        );
+        $candidateSearchFilterParamHolder->setMethodOfApplication(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_METHOD_OF_APPLICATION
+            )
+        );
+        $candidateSearchFilterParamHolder->setCandidateName(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_CANDIDATE_NAME
+            )
+        );
+        $candidates = $this->getCandidateService()->getCandidateDao()->getCandidatesList(
+            $candidateSearchFilterParamHolder
+        );
+        return new EndpointCollectionResult(
+            $this->getModelClass(),
+            $candidates,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => count($candidates)])
+        );
+    }
+
+    /**
+     * @return string|null
+     */
+    private function getStatusFilterParam(): ?string
+    {
+        $state = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_STATUS
+        );
+        return $state ? self::STATUS_MAP[$state] : null;
+    }
+
+    /**
+     * @return DateTime|null
+     */
+    private function getFromDateFilterParam(): ?DateTime
+    {
+        return $this->getRequestParams()->getDateTimeOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_FROM_DATE
+        );
+    }
+
+    /**
+     * @return void
+     * @throws BadRequestException
+     */
+    private function validateDateFilterParams()
+    {
+        $fromDate = $this->getFromDateFilterParam();
+        $toDate = $this->getToDateFilterParam();
+
+        if (($fromDate && $toDate) && $fromDate >= $toDate) {
+            throw $this->getBadRequestException("From Date Should Be Earlier Than To Date");
+        }
+    }
+
+    /**
+     * @return DateTime|null
+     */
+    private function getToDateFilterParam(): ?DateTime
+    {
+        return $this->getRequestParams()->getDateTimeOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_TO_DATE
+        );
+    }
+
+    /**
+     * @return string
+     */
+    protected function getModelClass(): string
+    {
+        $model = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_MODEL,
+            self::MODEL_DEFAULT,
+        );
+        return self::MODEL_MAP[$model];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::FILTER_CANDIDATE_ID,
```

Comment:
who can access this API? based on that may have to control access

Reply:
Need to add permissions based on Interviewer and hiring manager